src/main/kotlin/com/exactpro/th2/rptdataprovider/entities/requests/SseMessageSearchRequest.kt

```diff
@@ -164,6 +164,14 @@ data class SseMessageSearchRequest<RM, PM>(
         }
     }
 
+    private fun checkLookupLimitDays() {
+        if (lookupLimitDays != null && endTimestamp != null) {
+            throw InvalidRequestException(
+                "endTimestamp: $endTimestamp must be null if lookupLimitDays: $lookupLimitDays isn't null"
+            )
+        }

```

Comment:
question: Why can't we just prioritize the `endTimestamp` over `lookupLimitDays` in this case? If the end timestamp is explicitly specified it is obvious that the user wants to use it.

Reply:
Also, maybe we could pre-compute the `endTimestamp` in the constructor if `lookupLimitDays` parameter was provided. It should simplify the further use of SseMessageSearchRequest